pkg/kubelet/kubelet.go

```diff
@@ -428,7 +428,7 @@ func NewMainKubelet(
 			imageBackOff,
 			serializeImagePulls,
 			enableCustomMetrics,
-			klet.hairpinMode == componentconfig.HairpinVeth,
+			klet.hairpinMode == componentconfig.HairpinVeth && networkPluginName != "kubenet",
```

Comment:
Ugh, could we just rework things to pass the full hairpin mode flag through to the plugins and let them do what they want instead of special casing stuff?